runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -418,13 +418,13 @@ public void onChatMessage(ChatMessage event)
 		}
 
 		Matcher mProgress = CHAT_GEM_PROGRESS_MESSAGE.matcher(chatMsg);
+
 		if (!mProgress.find())
 		{
 			return;
 		}
 		String taskName = mProgress.group(1);
-		int amount = Integer.parseInt(mProgress.group(2));
-
+		int amount = Integer.parseInt(mProgress.group(3));
```

Comment:
Don't change this. Make the "in the Wilderness" group a non-capturing group in the regex.

Reply:
Im really shit with regex, how do i do this?

Reply:
https://www.regular-expressions.info/brackets.html#noncap

```
private static final Pattern CHAT_GEM_PROGRESS_MESSAGE = Pattern.compile("You're assigned to kill (.*?)(?:in the Wilderness)?; only (\\d*) more to go\\.");
```

Reply:
Thanks!